codex/utils/timer.nim

```diff
@@ -35,6 +35,9 @@ proc timerLoop(timer: Timer) {.async: (raises: []).} =
       await sleepAsync(timer.interval)
   except CancelledError:
     discard # do not propagate as timerLoop is asyncSpawned
+  except CatchableError as err:
+    error "CatchableError in timer loop", name = timer.name, msg = err.msg
```

Comment:
How did this compile before if there were no leaking `CatchableErrors` (due to `{.async: (raises: []).}`? If it is now not compiling, where were new exceptions introduced?